### PR TITLE
:sparkles: shared states

### PR DESCRIPTION
**Pull request by Molvilada**
Correcciones para accesibilidad 

## Summary by CodeRabbit

- **New Features**
	- Introduced a `StateMediator` for centralized state management across various view models.
	- Added a new loading state indicator and error/success state management in the UI.
	- Added a new composable function `InitalPagesTitle` for improved title display.

- **Bug Fixes**
	- Removed unnecessary parameters related to error handling from screens, simplifying the UI interaction.

- **Refactor**
	- Updated view models to utilize the `StateMediator` for managing loading and error states, enhancing separation of concerns.
	- Streamlined function signatures across multiple screens to improve clarity and maintainability.

- **Documentation**
	- Added unit tests for `StateMediator` to validate loading, error, and success state management.

### CODE REVIEW

File: app/src/main/java/io/capibaras/abcall/viewmodels/NavigationViewModel.kt [hunk 1]
```diff
@@ -14,30 +14,26 @@ import io.capibaras.abcall.ui.viewmodels.SuccessUIState
 import kotlinx.coroutines.launch
 
 class NavigationViewModel(
+    private val sharedViewModel: SharedViewModel,
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/NavigationViewModel.kt]: SharedViewModel lo estoy utilizando para manejar estados globales que me ayuden con los snackbar y los loading

File: app/src/main/java/io/capibaras/abcall/viewmodels/NavigationViewModel.kt [hunk 2]
```diff
@@ -14,30 +14,26 @@ import io.capibaras.abcall.ui.viewmodels.SuccessUIState
...
         checkUserSession()
         viewModelScope.launch {
             logoutManager.logoutState.collect { state ->
-                errorUIState = if (state.isExpiredToken) {
-                    ErrorUIState.Error(R.string.expired_token)
+                if (state.isExpiredToken) {
+                    sharedViewModel.setErrorState(ErrorUIState.Error(R.string.expired_token))
                 } else {
-                    ErrorUIState.NoError
+                    sharedViewModel.setErrorState(ErrorUIState.NoError)
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/NavigationViewModel.kt]: ¿Como sería podría utilizar un patron madiador para estados compartidos?

File: app/src/main/java/io/capibaras/abcall/di/AppModule.kt [hunk 3]
```diff
@@ -88,14 +89,28 @@ val appModule = module {
     single { CompanyRepository(get<CompanyDAO>(), get<CompanyService>(), get<SharedPreferences>()) }
     single { UsersRepository(get<UsersService>(), get<UserDAO>(), get<CompanyRepository>()) }
 
+    single { SharedViewModel() }
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/di/AppModule.kt]: Como maneja estados compartidos necesito que sea single y no viewmodel

File: app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt [hunk 4]
```diff
@@ -122,65 +117,52 @@ class SignUpViewModel(
...
-            isLoading = true
+            sharedViewModel.setLoadingState(true)
             try {
                 companies = companyRepository.getCompanies(forceUpdate)
             } catch (e: IOException) {
-                errorUIState = ErrorUIState.Error(R.string.error_network)
+                sharedViewModel.setErrorState(ErrorUIState.Error(R.string.error_network))
             } catch (e: Exception) {
-                errorUIState = ErrorUIState.Error(R.string.error_get_companies)
+                sharedViewModel.setErrorState(ErrorUIState.Error(R.string.error_get_companies))
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt]: No necesito que me pongas más comentarios sobre poner Log en catch

File: app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt [hunk 5]
```diff
@@ -122,65 +117,52 @@ class SignUpViewModel(
...
-                        ErrorUIState.Error(message = message)
+                        sharedViewModel.setErrorState(ErrorUIState.Error(message = message))
                     }
                 }
             } catch (e: IOException) {
-                errorUIState = ErrorUIState.Error(R.string.error_network)
+                sharedViewModel.setErrorState(ErrorUIState.Error(R.string.error_network))
             } catch (e: Exception) {
-                errorUIState = ErrorUIState.Error(R.string.error_create_user)
+                sharedViewModel.setErrorState(ErrorUIState.Error(R.string.error_create_user))
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt]: No necesito que me pongas más comentarios sobre poner Log en catch

**Review by VivianaOj — APPROVED** [review]

**Review by VivianaOj — APPROVED** [review]

**Pull request merged**